lib/archethic.ex

```diff
@@ -201,6 +242,43 @@ defmodule Archethic do
     end
   end
 
+  defp do_get_transaction_chain({true, paging_address}, address) when is_binary(paging_address) do
+    chain_from_local =
+      Task.async(fn ->
+        case fetch_transaction_chain_locally(paging_address) do
+          {:ok, chain_page} -> chain_page
+          _ -> []
+        end
+      end)
+
+    chain_from_network =
+      Task.async(fn ->
+        case get_transaction_chain_by_paging_address(address, paging_address) do
+          {:ok, chain_page} -> chain_page
+          _ -> []
+        end
+      end)
+
+    txn_list = (Task.await(chain_from_local) ++ Task.await(chain_from_network)) |> elem(1)
+    {:ok, txn_list}
+  end
+
+  defp fetch_transaction_chain_locally(paging_address) do
+    try do
+      nodes = [P2P.get_node_info()]
+
+      chain_page =
+        paging_address
+        |> TransactionChain.stream_remotely(nodes)
```

Comment:
This function aims to get the local transaction chain, but you get it remotely so it's not the local chain.
You should use `TransactionChain.get`

Reply:
I am querying the same nodes via P2P.message abstraction. 
It twill be more secure rather than directly querying the DB.